### PR TITLE
ERL-467: nemos-images-reference-*: *: add checkbox-erlangen-classic snap

### PR DESCRIPTION
**Pull request by IsaacJT**
Add the custom Checkbox snap created for the project. This contains all of the project-specific test suites that we require.

### CODE REVIEW

**Comment by IsaacJT** [comment]
Requires https://github.com/Elektrobit/nemos-images/pull/54

**Comment by IsaacJT** [comment]
Ready for review @ahresse @m-kat

**Review by ahresse — APPROVED** [review]
Works for me on development profile for mantic reference image.

```
# snap list checkbox-erlangen-classic
Name                       Version  Rev  Tracking       Publisher            Notes
checkbox-erlangen-classic  0.1dev   2    latest/stable  ce-certification-qa  classic
```

Thank you Isaac for adding spread tests.

**Comment by IsaacJT** [comment]
CI fixed

File: nemos-images-reference-lunar/qemu-amd64/config.sh [hunk 1]
```diff
@@ -52,16 +52,22 @@ for profile in ${kiwi_profiles//,/ }; do
         # This requires network access to the Snap Store in Kiwi.
         mkdir -p /var/lib/snapd/seed
         echo "snaps": > /var/lib/snapd/seed/seed.yaml
-        for snap in snapd checkbox22 checkbox core22; do
-            snap download $snap
+        for snap in snapd checkbox22 checkbox checkbox-erlangen-classic core22; do
```
Comment (zongminl) [inline nemos-images-reference-lunar/qemu-amd64/config.sh:55]: Let's remove the `checkbox` snap in this list, we don't need both checkbox and checkbox-erlangen-classic in NemOS image

File: nemos-images-reference-lunar/qemu-amd64/config.sh [hunk 2]
```diff
@@ -52,16 +52,22 @@ for profile in ${kiwi_profiles//,/ }; do
...
             cat >> /var/lib/snapd/seed/seed.yaml << EOF
   - name: ${snap}
-    channel: latest/stable
+    channel: ${CHANNEL}
     file: $(ls ${snap}_*.snap)
 EOF
-            # Checkbox snap requires classic confinement mode
-            if [ "${snap}" = "checkbox" ]; then
+            # Checkbox snaps require classic confinement mode
+            if [ "${snap}" = "checkbox" ] || [ "${snap}" = "checkbox-erlangen-classic" ]; then
```
Comment (zongminl) [inline nemos-images-reference-lunar/qemu-amd64/config.sh:70]: `if [ "${snap}" = "checkbox-erlangen-classic" ]; then` would be good enough

File: nemos-images-reference-lunar/qemu-arm64/config.sh [hunk 3]
```diff
@@ -52,16 +52,22 @@ for profile in ${kiwi_profiles//,/ }; do
         # This requires network access to the Snap Store in Kiwi.
         mkdir -p /var/lib/snapd/seed
         echo "snaps": > /var/lib/snapd/seed/seed.yaml
-        for snap in snapd checkbox22 checkbox core22; do
-            snap download $snap
+        for snap in snapd checkbox22 checkbox checkbox-erlangen-classic core22; do
```
Comment (zongminl) [inline nemos-images-reference-lunar/qemu-arm64/config.sh:55]: Let's remove the `checkbox` snap in this list, we don't need both checkbox and checkbox-erlangen-classic in NemOS image

File: nemos-images-reference-lunar/qemu-arm64/config.sh [hunk 4]
```diff
@@ -52,16 +52,22 @@ for profile in ${kiwi_profiles//,/ }; do
...
             cat >> /var/lib/snapd/seed/seed.yaml << EOF
   - name: ${snap}
-    channel: latest/stable
+    channel: ${CHANNEL}
     file: $(ls ${snap}_*.snap)
 EOF
-            # Checkbox snap requires classic confinement mode
-            if [ "${snap}" = "checkbox" ]; then
+            # Checkbox snaps require classic confinement mode
+            if [ "${snap}" = "checkbox" ] || [ "${snap}" = "checkbox-erlangen-classic" ]; then
```
Comment (zongminl) [inline nemos-images-reference-lunar/qemu-arm64/config.sh:70]: `if [ "${snap}" = "checkbox-erlangen-classic" ]; then` would be good enough

File: nemos-images-reference-lunar/s32g274ardb2/config.sh [hunk 5]
```diff
@@ -53,16 +53,22 @@ for profile in ${kiwi_profiles//,/ }; do
         # This requires network access to the Snap Store in Kiwi.
         mkdir -p /var/lib/snapd/seed
         echo "snaps": > /var/lib/snapd/seed/seed.yaml
-        for snap in snapd checkbox22 checkbox core22; do
-            snap download $snap
+        for snap in snapd checkbox22 checkbox checkbox-erlangen-classic core22; do
```
Comment (zongminl) [inline nemos-images-reference-lunar/s32g274ardb2/config.sh:56]: Let's remove the `checkbox` snap in this list, we don't need both checkbox and checkbox-erlangen-classic in NemOS image

File: nemos-images-reference-lunar/s32g274ardb2/config.sh [hunk 6]
```diff
@@ -53,16 +53,22 @@ for profile in ${kiwi_profiles//,/ }; do
...
             cat >> /var/lib/snapd/seed/seed.yaml << EOF
   - name: ${snap}
-    channel: latest/stable
+    channel: ${CHANNEL}
     file: $(ls ${snap}_*.snap)
 EOF
-            # Checkbox snap requires classic confinement mode
-            if [ "${snap}" = "checkbox" ]; then
+            # Checkbox snaps require classic confinement mode
+            if [ "${snap}" = "checkbox" ] || [ "${snap}" = "checkbox-erlangen-classic" ]; then
```
Comment (zongminl) [inline nemos-images-reference-lunar/s32g274ardb2/config.sh:71]: `if [ "${snap}" = "checkbox-erlangen-classic" ]; then` would be good enough

File: nemos-images-reference-mantic/qemu-amd64/config.sh [hunk 7]
```diff
@@ -52,16 +52,22 @@ for profile in ${kiwi_profiles//,/ }; do
         # This requires network access to the Snap Store in Kiwi.
         mkdir -p /var/lib/snapd/seed
         echo "snaps": > /var/lib/snapd/seed/seed.yaml
-        for snap in snapd checkbox22 checkbox core22; do
-            snap download $snap
+        for snap in snapd checkbox22 checkbox checkbox-erlangen-classic core22; do
```
Comment (zongminl) [inline nemos-images-reference-mantic/qemu-amd64/config.sh:55]: Let's remove the `checkbox` snap in this list, we don't need both checkbox and checkbox-erlangen-classic in NemOS image

File: nemos-images-reference-mantic/qemu-amd64/config.sh [hunk 8]
```diff
@@ -52,16 +52,22 @@ for profile in ${kiwi_profiles//,/ }; do
...
             cat >> /var/lib/snapd/seed/seed.yaml << EOF
   - name: ${snap}
-    channel: latest/stable
+    channel: ${CHANNEL}
     file: $(ls ${snap}_*.snap)
 EOF
-            # Checkbox snap requires classic confinement mode
-            if [ "${snap}" = "checkbox" ]; then
+            # Checkbox snaps require classic confinement mode
+            if [ "${snap}" = "checkbox" ] || [ "${snap}" = "checkbox-erlangen-classic" ]; then
```
Comment (zongminl) [inline nemos-images-reference-mantic/qemu-amd64/config.sh:70]: `if [ "${snap}" = "checkbox-erlangen-classic" ]; then` would be good enough

File: nemos-images-reference-mantic/qemu-arm64/config.sh [hunk 9]
```diff
@@ -52,16 +52,22 @@ for profile in ${kiwi_profiles//,/ }; do
         # This requires network access to the Snap Store in Kiwi.
         mkdir -p /var/lib/snapd/seed
         echo "snaps": > /var/lib/snapd/seed/seed.yaml
-        for snap in snapd checkbox22 checkbox core22; do
-            snap download $snap
+        for snap in snapd checkbox22 checkbox checkbox-erlangen-classic core22; do
```
Comment (zongminl) [inline nemos-images-reference-mantic/qemu-arm64/config.sh:55]: Let's remove the `checkbox` snap in this list, we don't need both checkbox and checkbox-erlangen-classic in NemOS image

File: nemos-images-reference-mantic/qemu-arm64/config.sh [hunk 10]
```diff
@@ -52,16 +52,22 @@ for profile in ${kiwi_profiles//,/ }; do
...
             cat >> /var/lib/snapd/seed/seed.yaml << EOF
   - name: ${snap}
-    channel: latest/stable
+    channel: ${CHANNEL}
     file: $(ls ${snap}_*.snap)
 EOF
-            # Checkbox snap requires classic confinement mode
-            if [ "${snap}" = "checkbox" ]; then
+            # Checkbox snaps require classic confinement mode
+            if [ "${snap}" = "checkbox" ] || [ "${snap}" = "checkbox-erlangen-classic" ]; then
```
Comment (zongminl) [inline nemos-images-reference-mantic/qemu-arm64/config.sh:70]: `if [ "${snap}" = "checkbox-erlangen-classic" ]; then` would be good enough

File: nemos-images-reference-mantic/s32g274ardb2/config.sh [hunk 11]
```diff
@@ -53,16 +53,22 @@ for profile in ${kiwi_profiles//,/ }; do
         # This requires network access to the Snap Store in Kiwi.
         mkdir -p /var/lib/snapd/seed
         echo "snaps": > /var/lib/snapd/seed/seed.yaml
-        for snap in snapd checkbox22 checkbox core22; do
-            snap download $snap
+        for snap in snapd checkbox22 checkbox checkbox-erlangen-classic core22; do
```
Comment (zongminl) [inline nemos-images-reference-mantic/s32g274ardb2/config.sh:56]: Let's remove the `checkbox` snap in this list, we don't need both checkbox and checkbox-erlangen-classic in NemOS image

File: nemos-images-reference-mantic/s32g274ardb2/config.sh [hunk 12]
```diff
@@ -53,16 +53,22 @@ for profile in ${kiwi_profiles//,/ }; do
...
             cat >> /var/lib/snapd/seed/seed.yaml << EOF
   - name: ${snap}
-    channel: latest/stable
+    channel: ${CHANNEL}
     file: $(ls ${snap}_*.snap)
 EOF
-            # Checkbox snap requires classic confinement mode
-            if [ "${snap}" = "checkbox" ]; then
+            # Checkbox snaps require classic confinement mode
+            if [ "${snap}" = "checkbox" ] || [ "${snap}" = "checkbox-erlangen-classic" ]; then
```
Comment (zongminl) [inline nemos-images-reference-mantic/s32g274ardb2/config.sh:71]: `if [ "${snap}" = "checkbox-erlangen-classic" ]; then` would be good enough

**Review by zongminl — CHANGES_REQUESTED** [review]
Some changes required, please see my inline comments

File: nemos-images-reference-lunar/qemu-amd64/config.sh (hunk 1 above)
Comment (ahresse) [reply nemos-images-reference-lunar/qemu-amd64/config.sh:55]: Should we remove both `checkbox22` and `checkbox` or only `checkbox`?

Comment (zongminl) [reply nemos-images-reference-lunar/qemu-amd64/config.sh:55]: We should still keep `checkbox22`, it provides all the required checkbox-runtime

File: nemos-images-reference-lunar/qemu-amd64/config.sh (hunk 2 above)
Comment (ahresse) [reply nemos-images-reference-lunar/qemu-amd64/config.sh]: https://github.com/Elektrobit/nemos-images/pull/55/commits/93053e6b6054c4c9fdb49869bb15599519a8657a

File: nemos-images-reference-lunar/qemu-arm64/config.sh (hunk 3 above)
Comment (ahresse) [reply nemos-images-reference-lunar/qemu-arm64/config.sh]: https://github.com/Elektrobit/nemos-images/pull/55/commits/93053e6b6054c4c9fdb49869bb15599519a8657a

File: nemos-images-reference-lunar/qemu-arm64/config.sh (hunk 4 above)
Comment (ahresse) [reply nemos-images-reference-lunar/qemu-arm64/config.sh]: https://github.com/Elektrobit/nemos-images/pull/55/commits/93053e6b6054c4c9fdb49869bb15599519a8657a

File: nemos-images-reference-lunar/s32g274ardb2/config.sh (hunk 5 above)
Comment (ahresse) [reply nemos-images-reference-lunar/s32g274ardb2/config.sh]: https://github.com/Elektrobit/nemos-images/pull/55/commits/93053e6b6054c4c9fdb49869bb15599519a8657a

File: nemos-images-reference-mantic/qemu-amd64/config.sh (hunk 7 above)
Comment (ahresse) [reply nemos-images-reference-mantic/qemu-amd64/config.sh]: https://github.com/Elektrobit/nemos-images/pull/55/commits/93053e6b6054c4c9fdb49869bb15599519a8657a

File: nemos-images-reference-mantic/qemu-amd64/config.sh (hunk 8 above)
Comment (ahresse) [reply nemos-images-reference-mantic/qemu-amd64/config.sh]: https://github.com/Elektrobit/nemos-images/pull/55/commits/93053e6b6054c4c9fdb49869bb15599519a8657a

File: nemos-images-reference-mantic/qemu-arm64/config.sh (hunk 9 above)
Comment (ahresse) [reply nemos-images-reference-mantic/qemu-arm64/config.sh]: https://github.com/Elektrobit/nemos-images/pull/55/commits/93053e6b6054c4c9fdb49869bb15599519a8657a

File: nemos-images-reference-mantic/qemu-arm64/config.sh (hunk 10 above)
Comment (ahresse) [reply nemos-images-reference-mantic/qemu-arm64/config.sh]: https://github.com/Elektrobit/nemos-images/pull/55/commits/93053e6b6054c4c9fdb49869bb15599519a8657a

**Comment by IsaacJT** [comment]
LGTM, thanks for the additions

**Review by zongminl — APPROVED** [review]
+1, thank you

**Pull request merged**